### DISCUSSION TITLE
Example projects where stubs override builtin annotations

### DISCUSSION THREAD

**Discussion by tgross35** [PyO3/pyo3#2537] [Questions]
Hello everyone,

I am attempting to follow the ["if you need other python files" layout in the docs appendix](https://pyo3.rs/latest/python_typing_hints.html?highlight=stub#if-you-need-other-python-files), but have been completely unsuccessful getting the stub file to take precedence over the builtin definition. That is, `myfunction.__doc__` returns the rust docstring rather than what's in the `.pyi`. This also means that `sphinx` cannot correctly generate autodocs.

So, is there a good example project that demonstrates how to correctly use stub files?

For reference, my existing nonworking project is located here: <https://github.com/pluots/stringmetrics/tree/5e494f0969b2b6dbc189696fc59365c8c2272032/crates/stringmetrics_py>, which has an identical layout to what is in the docs.

**Comment by messense** [comment]
I don't think sphinx autodoc supports `.pyi` file right now, there is a PR to add the feature, see https://github.com/sphinx-doc/sphinx/pull/4824.

You could try [sphinx-autoapi](https://github.com/readthedocs/sphinx-autoapi).
